src/platform/datapath_raw_socket.c

```diff
@@ -372,27 +369,72 @@ CxPlatResolveRoute(
     // Map the next hop IP address to a link-layer address.
     //
     MIB_IPNET_ROW2 IpnetRow = {0};
-    IpnetRow.InterfaceLuid = IpforwardRow.InterfaceLuid;
+    IpnetRow.InterfaceIndex = IpforwardRow.InterfaceIndex;
```

Comment:
Why change? I believe I used Luid because it was more efficient than index, which eventually has to be converted to Luid anyways.

Reply:
Luid is indeed more efficient. I changed it because it would make the plumbing easier. We now store interface pointer in route object and interface object has an ifindex field. We need interface object anyway for assigning DP queue. So, we don't need to plumb Luid for route operation and CxPlatQueryRoute doesn't need to output a Luid.

Reply:
fixed